beacon-chain/rpc/validator_server.go

```diff
@@ -132,29 +131,42 @@ func (vs *ValidatorServer) ValidatorPerformance(
 //	2.) The shard to which the committee is assigned.
 //	3.) The slot at which the committee is assigned.
 //	4.) The bool signaling if the validator is expected to propose a block at the assigned slot.
-func (vs *ValidatorServer) CommitteeAssignment(
-	ctx context.Context,
-	req *pb.CommitteeAssignmentsRequest) (*pb.CommitteeAssignmentResponse, error) {
-	beaconState, err := vs.beaconDB.HeadState(ctx)
+func (vs *ValidatorServer) CommitteeAssignment(ctx context.Context, req *pb.AssignmentRequest) (*pb.AssignmentResponse, error) {
+	s, err := vs.beaconDB.HeadState(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
 	}
 
-	var assignments []*pb.CommitteeAssignmentResponse_CommitteeAssignment
-	activeKeys := vs.filterActivePublicKeys(beaconState, req.PublicKeys)
-	for _, pk := range activeKeys {
+	validatorIndexMap := stateutils.ValidatorIndexMap(s)
+	var assignments []*pb.AssignmentResponse_Assignment
+
+	for _, pk := range req.PublicKeys {
 		if ctx.Err() != nil {
 			return nil, ctx.Err()
 		}
+		idx, ok := validatorIndexMap[bytesutil.ToBytes32(pk)]
+		// Default assignment for every validator
+		assignment := &pb.AssignmentResponse_Assignment{
```

Comment:
is there a better name for this message ? 
`&pb.AssignmentResponse_Assignment` is not the best name